icubam/map_builder.py

```diff
@@ -43,10 +43,18 @@ def to_map_data(self, tree, level):
     result.sort(key=lambda x: x['lat'], reverse=True)
     return result
 
-  def prepare_jsons(self, user_id=None, center_icu=None, level='dept'):
-    # TODO(olivier): all the icus on map ? or restrict somehow ?
+  def prepare_jsons(
+    self,
+    user_id: Optional[int] = None,
+    center_icu: Optional[store.ICU] = None,
+    regions: Optional[List[int]] = None,
+    level: str = 'dept'
+  ):
     tree = icu_tree.ICUTree()
-    tree.add_many(self.db.get_icus(), self.db.get_latest_bed_counts())
+    icus = self.db.get_icus()
+    if regions is not None and len(regions) > 0:
+      icus = [icu for icu in icus if icu.region_id in regions]
+    tree.add_many(icus, self.db.get_latest_bed_counts())
```

Comment:
I there are no regions, it would use ICUs from all regions. Is that desirable? I would have expected the above condition to only be `regions is not None` same as in the other file.

Reply:
indeed this is a bit of the same here, since regions in the store comes as a list (maybe empty), so we cannot set from the backoffice or the store the regions to be None. On the backoffice we choose not to have region restrictions and hence get an empty list.